### DISCUSSION TITLE
"Redash automatic deployment does not work after setting up DockerHub credentials."

### DISCUSSION THREAD

**Discussion by Jengro777** [RedashCommunity/redash#50] [Q&A] [answered]
Can someone help me？

![image](https://github.com/RedashCommunity/redash/assets/35259040/d866aad0-d7d6-461c-8d4e-e478362f5fee)
![image](https://github.com/RedashCommunity/redash/assets/35259040/0523606d-6a35-4681-b46b-046b1552c3e2)

------

![image](https://github.com/RedashCommunity/redash/assets/35259040/a1ae3e65-543a-4b3d-bf07-6b9530f0d61b)
![image](https://github.com/RedashCommunity/redash/assets/35259040/34f7aae9-ae86-412f-9ddb-239e68060cea)

**Comment by justinclift** [comment]
@hazmeister Any ideas? :smile:

**Reply by Jengro777** [reply]
> @hazmeister Any ideas? 😄

I want to complete the deployment of my own branch test before pushing it to the community branch. However, the automatic construction of the docker image on my end has not been successful. I have confirmed that my account and password settings are correct.

**Reply by Jengro777** [reply]
I have correctly set up my DockerHub account, password, and key. Why does it still tell me that the DockerHub password is incorrect?

**Reply by hazmeister** [reply]
Looks good, I think I had the same trouble.  In order to use the environment secrets, you need to use the environment from your workflow file:
https://github.com/Avey777/RedashCommunity-redash/actions/runs/5253591255/workflow

Like here https://github.com/RedashCommunity/redash/blob/master/.github/workflows/docker-preview.yml#L10

**Reply by Jengro777** [reply]
![image](https://github.com/RedashCommunity/redash/assets/35259040/91aba0b2-ae6d-45d2-be18-9d3976dacfa7)

**Reply by hazmeister** [reply]
Sorry, missed that.  We're using a token for login- did you create one [here](https://hub.docker.com/settings/security)?

**Reply by Jengro777** [reply]
![image](https://github.com/RedashCommunity/redash/assets/35259040/bcf95ccb-f1ec-4e12-b24e-ace511982ebc)
![image](https://github.com/RedashCommunity/redash/assets/35259040/708f165c-cef7-48d5-94f1-80cb2208e82a)

**Reply by Jengro777** [reply]
Can I invite you to join my branch and help me?

**Reply by hazmeister** [reply]
![image](https://github.com/RedashCommunity/redash/assets/11080542/c8ccca93-c5f0-450b-9aee-42465c5ee695)
These should all be prefixed with `DOCKERHUB`

**Reply by Jengro777** [reply]
Yes, I have confirmed it many times.

**Reply by hazmeister** [reply, marked as answer]
but in your screenshot, you have `DOCKER_`USERNAME, not `DOCKER_HUB_`USERNAME

**Reply by Jengro777** [reply]
I understand what you mean.

**Reply by Jengro777** [reply]
I will try it right away.

**Reply by Jengro777** [reply]
Thank you so much! It has started working now.

**Reply by hazmeister** [reply]
No worries, the human brain is really bad at recognising typos- if the front and back look good, your brain jumps straight to this is fine! Glad it's working :)

**Reply by justinclift** [reply]
Awesome, *thanks heaps* @hazmeister! :smile:
